src/ebonite/client/base.py

```diff
@@ -144,42 +145,59 @@ def get_model(self, model_name: str, task: TaskVar, project: ProjectVar = None,
             model.load()
         return model
 
-    def build_image(self, name: str, model: Model, server: Server = None, environment: RuntimeEnvironment = None,
+    def build_image(self, name: str, obj, task: Task, server: Server = None, environment: RuntimeEnvironment = None,
                     debug=False, **kwargs) -> Image:
         """
         Builds image of model service and stores it to repository
 
         :param name: name of image to build
-        :param model: model to wrap into service
+        :param obj: buildable object to wrap into service
+        :param task: task to put image into
         :param server: server to build image with
         :param environment: env to build for
         :param debug: flag to build debug image
         :param kwargs: additional kwargs for builder
         :return: :class:`~ebonite.core.objects.Image` instance representing built image
         """
-        if self.meta_repo.get_image_by_name(name, model) is not None:
+        from ebonite.core.analyzer.buildable import BuildableAnalyzer
+        if self.meta_repo.get_image_by_name(name, task) is not None:
             raise ExistingImageError(name)
         if server is None:
             server = self.get_default_server()
 
         if environment is None:
             environment = self.get_default_environment()
-        builder = environment.params.get_builder(name, model, server, debug, **kwargs)
+        buildable = BuildableAnalyzer.analyze(obj, server=server, debug=debug)
+        buildable.bind_meta_repo(self.meta_repo)
+        builder = environment.params.get_builder(name, buildable, **kwargs)
         image = builder.build()
-        image.model = model
+        image.task = task
         return self.meta_repo.create_image(image)
 
-    def get_image(self, name: str, model: Model) -> Image:
+    def get_pipeline(self, name: str, task: Task) -> Pipeline:
+        """
+        Load pipeline from repository
+
+        :param name: pipeline name to load
+        :param task: :py:class:`~ebonite.core.objects.Task` instance to load image from
+        :return: loaded :py:class:`~ebonite.core.objects.Pipeline` instance
+        """
+        return self.meta_repo.get_pipeline_by_name(name, task)
+
+    def delete_pipeline(self, pipeline: Pipeline):
```

Comment:
shouldn't it have cascade parameter and related functionality to remove associated models?

Reply:
no)))